Frontend/ui-customer/src/Flow.purs

```diff
@@ -783,10 +783,11 @@ homeScreenFlow = do
                                         let (RideAPIEntity ride) = fromMaybe dummyRideAPIEntity (resp.rideList !! 0)
                                         let finalAmount =  getFinalAmount (RideBookingRes resp)
                                         let differenceOfDistance = fromMaybe 0 contents.estimatedDistance - (fromMaybe 0 ride.chargeableRideDistance)
+                                        let nightSafetyFlow = not (fromMaybe true resp.hasNightIssue) && ((withinTimeRange "21:00:00" "06:00:00" (convertUTCtoISC(fromMaybe "" resp.rideStartTime) "HH:mm:ss")) || (withinTimeRange "21:00:00" "06:00:00" (convertUTCtoISC(fromMaybe "" resp.rideEndTime) "HH:mm:ss")))
```

Comment:
write clean code like - 

```purescript
  let nightSafetyFlow = 
    not (fromMaybe true resp.hasNightIssue) 
    && (isWithinTimeRange resp.rideStartTime
      || isWithinTimeRange resp.rideEndTime)
  .
  .
  .
  .
  where
    isWithinTimeRange :: Maybe String -> Boolean
    isWithinTimeRange = 
      maybe false (\time -> withinTimeRange "21:00:00" "06:00:00" $ convertUTCtoISC(fromMaybe "" time) "HH:mm:ss") 
``` 

also keep note that if time is `Nothing` I'm returning false. Is it expected? also cases should not go to convertUTCtoISC if string is empty because of fromMaybe so that's why added `maybe` function on top

Reply:
Got it